### PR TITLE
fix: Use ids compatible with CSS

### LINKED ISSUE #3409: Numeric element IDs in WASM are not compatible with CSS
## What would you like to be added:

Uno currently generates numeric IDs for elements, which however cannot be referenced by CSS rules normally:

For example this does not work:

    #423506::selection { ... } 

While this does:

    [id='423506']::selection { ... }

## Why is this needed:

To be able to style pseudo-selectors for elements.

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #3409

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Element IDs aren't compatible with CSS

## What is the new behavior?

Prefix the IDs with "uno-"

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/UIElement.wasm.cs [hunk 1]
```diff
@@ -94,7 +94,7 @@ public UIElement(string htmlTag, bool isSvg)
 			var type = GetType();
 
 			Handle = GCHandle.ToIntPtr(_gcHandle);
-			HtmlId = Handle;
+			HtmlId = $"uno-{Handle}";
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/UIElement.wasm.cs:97]: I'm really wondering why there is both `Handle` and `HtmlId` if they are the same? The properties are readonly and can't be changed after constructor execution, so they should always remain the same. What was the point of having both of them?

File: src/Uno.UI.Runtime.WebAssembly/Xaml/UIElementWasmExtensions.cs [hunk 2]
```diff
@@ -121,7 +121,7 @@ public static string ExecuteJavascript(this UIElement element, string jsCode)
 			var js = @$"
 (function(element) {{
 {jsCode}
-}})(Uno.UI.WindowManager.current.getView({element.HtmlId}));
+}})(Uno.UI.WindowManager.current.getView(""{element.HtmlId}""));
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.WebAssembly/Xaml/UIElementWasmExtensions.cs:124]: My concern about this PR is if I missed any places where quotes are needed. I hope I didn't miss, and hope coverage is enough to fail the CI if anything is missing.

**Comment by Youssef1313** [comment]
My best guess about the current CI failure is:

https://github.com/unoplatform/Uno.UITest/blob/9a9c6d22aebbc566cffcf1e930270531c8c2c795/src/Uno.UITest.Puppeteer/SeleniumApp.cs#L616

@MartinZikmund @davidjohnoliver How do we approach updates in both repositories and testing them against each other?

**Comment by jeromelaban** [comment]
This change will incur a performance cost because the ID is now a string that needs to be split and converted. We could investigate instead for using a different attribute on html element and see if the lookup performance is the same.

**Comment by Youssef1313** [comment]
> This change will incur a performance cost because the ID is now a string that needs to be split and converted. We could investigate instead for using a different attribute on html element and see if the lookup performance is the same.

I'm not following. I haven't done any splitting anywhere, so I don't get when/why does this happen. There is already "xamlhandle" attribute which is still numeric.

**Comment by jeromelaban** [comment]
> > This change will incur a performance cost because the ID is now a string that needs to be split and converted. We could investigate instead for using a different attribute on html element and see if the lookup performance is the same.
> 
> I'm not following. I haven't done any splitting anywhere, so I don't get when/why does this happen. There is already "xamlhandle" attribute which is still numeric.

Right, I read the code wrong for the splitting, but changing the id to a string is adding some performance cost. It needs to stay as a integer during the interop.

**Comment by Youssef1313** [comment]
> > > This change will incur a performance cost because the ID is now a string that needs to be split and converted. We could investigate instead for using a different attribute on html element and see if the lookup performance is the same.
> > 
> > 
> > I'm not following. I haven't done any splitting anywhere, so I don't get when/why does this happen. There is already "xamlhandle" attribute which is still numeric.
> 
> Right, I read the code wrong for the splitting, but changing the id to a string is adding some performance cost. It needs to stay as a integer during the interop.

So what should the next step be? Or does this suggest that the linked issue shouldn't be fixed?

**Comment by jeromelaban** [comment]
> So what should the next step be? Or does this suggest that the linked issue shouldn't be fixed?
Adding an `uno-` is interesting, but it should should be not be marshaled as is. It can be added when setting the `id` property on the javascript side, but based on the original `Handle` value.

Also, for all interop calls, the value to be passed needs to be the handle value, and for the element lookups, let's try to see if lookups on another property that is not "id" is as fast as the current lookup.

In all cases, this change is breaking so we can't merge it on master.

**Comment by carldebilly** [comment]
I'm still convinced this change is not required.

**Comment by jeromelaban** [comment]
I think we should not do this modification, and the original issue should be closed as well. Manipulating uno-generated elements using CSS is not a scenario we want to support, all things considered.

**Comment by Youssef1313** [comment]
Let's close the PR and the linked issue then.

**Pull request closed without merging**